sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -599,6 +599,61 @@ VirtualMachineResource virtualMachine = virtualMachineOperation.Value;
 
 Finally, as it can be seen here, from the resource group you can get the Virtual Machine collection and create a new one using the `VirtualMachineData` for the parameters.
 
+### Update a Virtual Machine
+
+**Create a new object of VirtualMachineData**
+```C#
+var virtualMachineDataToModify = new VirtualMachineData(AzureLocation.EastUS)
+{
+    NetworkProfile = new VirtualMachineNetworkProfile()
+    {
+        NetworkInterfaces =
+        {
+            new VirtualMachineNetworkInterfaceReference()
+            {
+                Id = networkInterface.Id,
+                Primary = false,
+            }
+        }
+    }
+};
+var virtualMachineModify = (await virtualMachines.CreateOrUpdateAsync(WaitUntil.Completed, virtualMachineName, virtualMachineDataToModify)).Value;
+```
```

Comment:
This is actually creating a whole new VM instance and override it if there is already one.

I believe the description here should be "Replace or create the virtual machine resource"
And here we should containing more things in this VirtualMachineData instantiation to emphasize that this way we need to have everything on the vm instead of like the patch way, we only need to contain the part we want to change.

Reply:
resolved